scripts/valhalla_build_config

```diff
@@ -579,7 +582,12 @@ def add_leaf_args(path, tree, leaves, parser):
       h = h[k]
     t = comma_separated_list if type(tree) == type([]) else (type(tree) if type(tree) != type(optional()) else tree.type)
     arg = '--' + path.replace('_', '-').replace('\0', '-')
-    parser.add_argument(arg, type = t if t != type(bool()) else parse_bool, help=h, default=tree)
+    if t == type(bool()):
+      t = parse_bool
+    elif t == type(list()):
+      t = parse_list
```

Comment:
moved the ternary up here to a normal if block so i could add the special case for lists. i suppose i could have used a dict as well, meh

Reply:
Is it possible to add a unit test for this change just in case?

Reply:
testing the config seems like a good idea. would need a few more changes though, no @keshavnandan ? e.g. `valhalla_build_config` can't be imported by a test module currently, missing the `__init__.py` in both the `valhalla` root and the `scripts` dir. IMO `valhalla_build_config` should be copied to `build/src/bindings/python/valhalla` during the build, then the python bindings could import, use and unit test it.

Reply:
at the moment we test it to make sure it works (ie returns non-zero). adding a unit test in this case would mean creating a whole new suite since we dont have one at all. im not against it, its just a decent chunk of work depending on how we do it.

regarding copying it to bindings, i humbly disagree. its not a binding to the c++ library, its just a helper script. that being said though, we've for a long time wanted to make a config object within the c++ library and get away from the unstructured ptree. if we did that work we could of course expose this to the python bindings and indeed have the same code shared all the way from the guts of the library, to the c++ unit tests, and into the python bindings (we'd probably make a c++ executable to generate configs to take the place of the python script though).

in the mean time, how about a cheaper test that is inline within the script? i can add another arg to the program that exorcises all of the args and all of their values and if something doesnt checkout it can `exit(1)` or `raise`. this program is obviously important but i feel like its not clear what the best way forward is yet.

Reply:
>  its not a binding to the c++ library, its just a helper script.

true. guess I'm more thinking of the bindings as a python package which is supposed to be imported in other projects/scripts. in that context it really makes a lot of sense to include the config script in the distribution, the package could create its config at runtime without relying on separate config files or system calls. depends a little how you look at it I guess..

reg inline tests, I think that's a painless sorta patchy good enough option :smile: